component/remote/http/http.go

```diff
@@ -39,14 +39,23 @@ type Arguments struct {
 	PollTimeout   time.Duration `river:"poll_timeout,attr,optional"`
 	IsSecret      bool          `river:"is_secret,attr,optional"`
 
+	Method  string    `river:"method,attr,optional"`
+	Headers []*Header `river:"header,block,optional"`
```

Comment:
This was my first approach!

But it would work well. I wrote a test for this code

```
remote.http "targets" {
  url = env("MY_TARGETS_URL")
  method = "GET"
  headers = {
    x-custom-header = ["value"],
  }
}
```

but it seems like `-` are not allowed in identifier. This does not work, too:

```
remote.http "targets" {
  url = env("MY_TARGETS_URL")
  method = "GET"
  headers = {
    "x-custom-header" = ["value"],
  }
}
```

Did no have the exact error message anymore. And the appending `,` is important, otherwise there is a syntax error. Then I switch to that block based approach.

Reply:
Other components like [prometheus.remote_write](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.remote_write/#endpoint-block) Use `` Headers map[string]string `river:"headers,attr,optional"` ``. 

For consistency, should we do the same here?

Reply:
```
        header {
            x-custom = ["x-value"]
        }
```
Results into: `expected block label, got -`

```
        header {
            'x-custom' = ["x-value"]
        }
```
Results into: `expected identifier, got STRING (and 2 more diagnostics)`

```
        header {
            "x-custom" = ["x-value"]
        }
```
Results into: `illegal single-quoted string; use double quotes`


> For consistency, should we do the same here?

I would recommend this, since there is no way to define header name with an `-`. By the way, this is allowed on HCL, but it seems like river behaves here differently.

Reply:
But please, move this discussion of out this PR

Reply:
I'm not sure why your River test failed, for me it looks it can be parsed correctly if using double-quoted strings.

```
# component/remote/http/http.go
	Headers map[string]string `river:"headers,attr,optional"`

# component/remote/http/http_test.go
        headers = {
		    "custom"   = "value",
			"x-custom" = "x-value1",
        }
```

Let's go with this approach for consistency with the components, and we're set.